float32 Global
float32 Safety
float32 Directness
float32 Smoothness

